boefjes/boefjes/plugins/kat_ssl_scan/main.py

```diff
@@ -1,15 +1,44 @@
 import subprocess
 
+TLS_CAPABLE_SERVICES = ("https", "ftps", "smtp", "smtps", "imaps", "pop3s", "ssh", "rpd")
+STARTTLS_CAPABLE_SERVICES = ("pop3", "ftp", "imap", "smtp", "mysql", "ldap", "xmpp")
+
 
 def run(boefje_meta: dict) -> list[tuple[set, bytes | str]]:
     input_ = boefje_meta["arguments"]["input"]
-    hostname = input_["hostname"]["name"]
+    hostname = None
+    if "hostname" in input_:
+        # we are dealing with a website
+        hostname = input_["hostname"]["name"]
+        ip = input_["ip_service"]["ip_port"]["address"]["address"]
+        ipfamily = input_["ip_service"]["ip_port"]["address"]["object_type"]
+        port = input_["ip_service"]["ip_port"]["ip"]
+        servicename = input_["ip_service"]["service"]["name"]
+    else:
+        # we are dealing with an IP-service
+        ip = input_["ip_port"]["address"]["address"]
+        ipfamily = input_["ip_port"]["address"]["object_type"]
```

Comment:
This results in `KeyError: 'object_type'`. You can copy the code from #4698 which has been tested and works.